src/routes/authentication-route.js

```diff
@@ -14,8 +15,19 @@ module.exports = function createAuthenticationRoute(repository) {
     return response.status(200).json({ token });
   }
 
+  async function comparePassword(request, response, next) {
+    const passwordDb = await repository.returnUser(request.body);
+    try {
+      const compareResult = await hashPassword.compare(request.body.password, passwordDb.user.dataValues.password);
```

Comment:
`passwordDb.user.dataValues.password` it seems that you overcomplicate things, usually you can access database values directly not through `dataValues`.

`user.password` should work if it is configured properly.